src/mumble/ServerHandler.cpp

```diff
@@ -454,6 +462,8 @@ void ServerHandler::message(unsigned int msgType, const QByteArray &qbaMsg) {
 			ConnectionPtr connection(cConnection);
 			if (!connection) return;
 
+			iInFlightTCPPings -= 1;
```

Comment:
Well, pings are restricted to ServerHandler: they're only ever sent via sendPing(), and the message handler lives in ServerHandler, (as opposed to Messages/MainWindow). So I think the current structure is OK.